frontend/src/contexts/Auth.tsx

```diff
@@ -0,0 +1,81 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+
+interface Auth {
+  isLoggedIn?: boolean;
+  logIn: (email: string, password: string) => Promise<Response>;
+  logOut: () => void;
+  getAuthorization: () => string;
+}
+
+const AuthContext = createContext<Auth | undefined>(undefined);
+
+interface Props {
+  children?: React.ReactNode;
+}
+
+interface ErrorResponse {
+  status: false;
+  error: string;
+}
+
+interface SuccessResponse {
+  status: true;
+  name: string;
+  token: string;
+}
+
+type Response = ErrorResponse | SuccessResponse;
+
+export const AuthProvider = ({ children }: Props) => {
+  const [isLoggedIn, setLoggedIn] = useState<boolean>();
+
+  const logIn = async (email: string, password: string) => {
+    const basicAuthToken = getBasicAuth(email, password);
+
+    const request = await fetch('http://localhost:5000/user/login/', {
+      method: 'POST',
+      headers: { Authorization: basicAuthToken }
+    });
+
+    const response = (await request.json()) as Response;
+    if (response.status) {
+      localStorage.setItem('name', response.name);
+      localStorage.setItem('token', response.token);
+      setLoggedIn(true);
+    }
+
+    return response;
+  };
+
+  const getBasicAuth = (email: string, password: string): string => {
+    const hash = btoa(`${email}:${password}`);
+    const token = `Basic ${hash}`;
+    return token;
+  };
```

Comment:
Variable names could use some work. I am aware that your input is a token and your output is another token, so names are going to get ugly either way.

Reply:
Don't try to pull a UNO Reverse card on me. In this context, the variable names are accurate.